src/sql/base/browser/ui/chart/chart.component.ts

```diff
@@ -0,0 +1,321 @@
+			(this.chart.canvas.parentNode as any).style.width = val;
+		}
+	}
+
+	private convert(val: azdata.ChartConfiguration): chartjs.ChartData {
+		const result: chartjs.ChartData = {
+			datasets: []
+		}
+
+		switch (this._type) {
+			case 'bar':
+			case 'horizontalBar': // should've been replaced with 'bar' by this point, but inlcuded case here for safety
+			case 'line':
+				{
+					const config = <azdata.BarChartConfiguration>val;
+					for (let set of config.datasets) {
+						result.datasets.push({
+							data: set.data.map(entry => typeof entry === 'number' ? entry : entry.x),
+							backgroundColor: set.backgroundColor,
+							borderColor: set.borderColor,
+							label: set.dataLabel
+						});
+					}
+
+					result.labels = config.labels;
+					break;
+				}
+			case 'pie':
+			case 'doughnut':
+				{
+					const config = <azdata.PieChartConfiguration>val;
+
+					result.datasets.push({
+						data: config.dataset.map(entry => typeof entry.value === 'number' ? entry.value : entry.value.x),
+						backgroundColor: config.dataset.map(entry => entry.backgroundColor),
+						borderColor: config.dataset.map(entry => entry.borderColor)
+					});
+
+					result.labels = config.dataset.map(val => val.dataLabel);
+					break;
+				}
+			case 'scatter':
+				{
+					const config = <azdata.ScatterplotConfiguration>val;
+
+					for (let set of config.datasets) {
+						result.datasets.push({
+							data: set.data.map(entry => [entry.x, entry.y]),
+							backgroundColor: set.backgroundColor,
+							borderColor: set.borderColor,
+							label: set.dataLabel
+						});
+					}
+
+					break;
+				}
+			case 'bubble':
+				{
+					const config = <azdata.BubbleChartConfiguration>val;
+
+					for (let set of config.datasets) {
+						result.datasets.push({
+							data: set.data.map(entry => ({ x: entry.x, y: entry.y, r: entry.r })),
+							backgroundColor: set.backgroundColor,
+							borderColor: set.borderColor,
+							label: set.dataLabel
+						});
+					}
+
+					break;
+				}
+			case 'polarArea':
+				{
+					const config = <azdata.PolarAreaChartConfiguration>val;
+
+					result.datasets.push({
+						data: config.dataset.map(entry => typeof entry.value === 'number' ? entry.value : entry.value.x),
+						backgroundColor: config.dataset.map(entry => entry.backgroundColor),
+						borderColor: config.dataset.map(entry => entry.borderColor)
+					});
+
+					result.labels = config.dataset.map(val => val.dataLabel);
+					break;
+				}
+			case 'radar':
+				{
+					const config = <azdata.RadarChartConfiguration>val;
+
+					for (let set of config.datasets) {
+						result.datasets.push({
+							data: set.data.map(entry => typeof entry === 'number' ? entry : entry.x),
+							backgroundColor: set.backgroundColor,
+							borderColor: set.borderColor,
+							label: set.dataLabel
+						});
+					}
+
+					result.labels = config.labels;
+					break;
+				}
+			default:
+				throw new Error(`Unsupported chart type: '${this._type}'`);
```

Comment:
Is this shown to the user? (and should be localized?)

Reply:
This error is for the logs; user shouldn't see it.